R/search.R

```diff
@@ -0,0 +1,76 @@
+
+#' Search for data products using the EarthData API
+#' @param short_name dataset short name e.g. ATL08
+#' @param version dataset version
+#' @param doi DOI for a dataset
+#' @param daac NSIDC or PODAAC
+#' @param provider particular to each DAAC, e.g. POCLOUD, LPDAAC etc.
+#' @param temporal c("yyyy-mm-dd", "yyyy-mm-dd")
+#' @param bounding_box c(lower_left_lon, lower_left_lat, upper_right_lon, upper_right_lat)
+#' @param page_size maximum number of results to return
+#' @param ... additional query parameters
+#' @export
+#' @return A response object of the returned search information
+#' @examplesIf interactive()
+#'
+#' resp <- edl_search(short_name = "MUR-JPL-L4-GLOB-v4.1",
+#'                    temporal = c("2020-01-01", "2021-12-31"))
+#'
+#' urls <- edl_extract_urls(resp)
+#'
+edl_search <- function(short_name = NULL,
+                       version = NULL,
+                       doi = NULL,
+                       daac = NULL,
+                       provider = NULL,
+                       temporal = NULL,
+                       bounding_box = NULL,
+                       page_size = 2000,
+                       ...) {
+
+  token <- earthdatalogin::edl_set_token(set_env_var = FALSE)
+
+  query <- list(short_name = short_name,
+                temporal = paste(temporal, collapse=","),
+                version = version,
+                doi = doi,
+                daac = daac,
+                provider = provider,
+                bounding_box = bounding_box,
+                page_size = page_size,
+                ...)
+  query <- purrr::compact(query)
+
+  url <- "https://cmr.earthdata.nasa.gov/search/granules"
+  resp <- httr::GET(url,
+                    query = query,
+                    httr::add_headers("Authorization"=paste("Bearer", token)))
```

Comment:
@betolink this bit is I think a pretty vanilla search of granules.  Not sure if I should be searching catalogs too?

Reply:
@cboettig if you go ahead with this Query builder, I'd say `concept-id` is the more deterministic way of referring to a dataset. Some datasets are in both cloud collections / on-prem collections. Looks good, maybe you could include `polgygon=` and internally convert the bounding boxes to polygons, that way you can support arbitrary geometries.